src/lib.rs

```diff
@@ -977,6 +977,21 @@ pub fn version_is_supported(version: u32) -> bool {
     }
 }
 
+macro_rules! add_frame {
```

Comment:
can this just be a function rather than a macro?

Reply:
I prefer to use macro because when it is a function there is many side effects (`&mut`), such as updating payload_len, left etc.